presto-main/src/main/java/com/facebook/presto/execution/scheduler/BucketNodeMap.java

```diff
@@ -39,6 +39,8 @@ public BucketNodeMap(ToIntFunction<Split> splitToBucket)
 
     public abstract boolean isDynamic();
 
+    public abstract boolean isCacheable();
```

Comment:
Why is `cacheable` a property for `BucketNodeMap`? Shouldn't this be a property for a node-bucket pair?